### PR TITLE
lifeline: allow override global rescue_time per worker

### PR DESCRIPTION
**Pull request by milmazz**
This PR adds the following two options to the `Oban.Plugins.Lifeline` plugin:

* `timeout` - to be able to modify the query timeouts
* `worker_overrides` - this is mainly to allow OSS developers to change the global `rescue_after` for more specific scenarios based on the worker name.

### CODE REVIEW

File: lib/oban/plugins/lifeline.ex [hunk 1]
```diff
@@ -37,6 +37,13 @@ defmodule Oban.Plugins.Lifeline do
   * `:rescue_after` — the maximum amount of time, in milliseconds, that a job may execute before
   being rescued. 60 minutes by default, and rescuing is performed once a minute.
 
+  * :`:worker_overrides` - allows to override the maximum amount of time, in
```
Comment (milmazz) [inline lib/oban/plugins/lifeline.ex:40]: * [ ] TODO: Add an example using this option

File: lib/oban/plugins/lifeline.ex [hunk 2]
```diff
@@ -143,20 +161,41 @@ defmodule Oban.Plugins.Lifeline do
...
+
+          worker_counts =
+            for {worker, rescue_after} <- state.worker_overrides do
+              time = DateTime.add(now, -rescue_after, :millisecond)
+              query = where(base, [j], j.attempted_at < ^time and j.worker == ^worker)
+              run_transitions(query, state)
+            end
+
+          time = DateTime.add(now, -state.rescue_after, :millisecond)
+          query = where(base, [j], j.attempted_at < ^time)
```
Comment (milmazz) [inline lib/oban/plugins/lifeline.ex:178]: * [ ] Probably worth excluding the worker names from this query

**Comment by sorentwo** [comment]
The basic lifeline plugin has a limited set of options intentionally, and worker overrides don't belong here. An optional query timeout is welcome (though unexpected for such a straightforward update query).

**Comment by milmazz** [comment]
> The basic lifeline plugin has a limited set of options intentionally, and worker overrides don't belong here.

Closing this in favor of a custom plugin for my use case, I will leave this here in case another developer finds it useful.

> An optional query timeout is welcome (though unexpected for such a straightforward update query).

Yeah, without additional filters such as the proposed `worker_overrides` it doesn't make sense to provide a `timeout` option for the current simple approach.

**Pull request closed without merging**